docs/v3/core/index.rst

```diff
@@ -349,6 +350,25 @@ terminology for a use case of reading from an array:
 .. image:: terminology-read.excalidraw.png
   :width: 600
 
+*Extension points*
+
+    Locations within a `metadata document_` where extension-related
+    metadata can be found. Current extension points are listed in the core spec,
+    e.g. `codecs`, `data_type`. See :ref:`extension points <extensions_section>` below.
+
+*Extensions*
+
+    Components defined in a `metadata document`_ to
+    configure how metadata are interpreted by implementations. These
+    components include codecs, data types, chunk key encodings, chunk grids and
+    storage transformers. See :ref:`extension points <extensions_section>` below.
```

Comment:
what is the difference between the "Extension points" and "Extensions" sections? It seems like we might only need one.

Reply:
This is trying to convey a class/instance style relationship:

| **Places to put things** | **Things to put there** |
|----------------------|---------------------|
| e.g. the codecs field | a particular codec |
| extension point      | extension           |
| type of extension    | extension           |
| extension            | ...what here? ...   |

Reply:
I don't think we should model "extension" as a class? Extensibility is property of a set of values. A set of values is extensible if new elements can be added. A set of values is not extensible if no new values can be added. We don't add anything to these definitions by thinking of "extension" as a class that has implementations / instances.

Reply:
I don't think we need two terms -- the definition of "extension point" is sufficient to convey "a field with a non-static set of values".